### PR TITLE
Delete expo-payments

### LINKED ISSUE #7555: Delete apps/expo-payments
Every app in the `apps/` folder adds extra time to the yarn install and git clone. We can/should move the IAP demo to NCL + Bare Expo. 

- `apps/expo-payments` is an ExpoKit app which is now deprecated in favor of the Bare Workflow.

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- No longer needed after https://github.com/expo/expo/pull/7540
- Improves yarn install times
- fix https://github.com/expo/expo/issues/7555

# How

- `rm -rf apps/expo-payments && yarn`

### CODE REVIEW

File: apps/expo-payments/android/app/google-services.json [hunk 1]
```diff
@@ -1,50 +0,0 @@
-{
-  "project_info": {
-    "project_id": "exponent-5dd6d",
-    "project_number": "367315174693",
-    "name": "Exponent Play Services"
-  },
-  "client": [
-    {
-      "client_info": {
-        "mobilesdk_app_id": "1:367315174693:android:f968efbb41d1fa7a",
```
Comment (sjchmiela) [inline apps/expo-payments/android/app/google-services.json]: Do you know what is this project and whether the "Custom Expo Development" Firebase project is configured appropriately for In App Purchases>

**Review by sjchmiela — APPROVED** [review]
Apart from that one concern I think it's ready to go! Thank you! 🙏

File: apps/expo-payments/android/app/google-services.json (hunk 1 above)
Comment (EvanBacon) [reply apps/expo-payments/android/app/google-services.json]: I don't there is any setup required in Firebase, this is just the services file that was created after ejecting to ExpoKit. The main concern was reusing the iOS payments as they can be more cumbersome to setup.

Comment (ide) [reply apps/expo-payments/android/app/google-services.json]: The Custom Expo Development project is the old payments project. It’s just renamed but the application ID is the same as the payments app ID.

**Pull request merged**